src/server/routes/api/document/pubmed/searchPubmed.js

```diff
@@ -51,6 +54,86 @@ const eSearchPubmed = ( term, opts ) => {
   .then( pubmedDataConverter );
 };
 
+/**
+ * Search for an article using Crossref and return a Pubmed-style result.
+ * 
+ * This approach uses Crossref to find the article's DOI.  The DOI is sent to PMC to find a
+ * PMID.  If the PMC query fails for any reason, a query to Pubmed Eutils is made using the 
+ * DOI as the search term.  For some articles, PMC fails.  For others, Eutils fails.  Using
+ * both gives good coverage.
+ * 
+ * @param {String} term The search term to find the article (e.g. title)
+ * @param {Object} opts Pubmed API options
+ * @returns {Object} A Pubmed search result object
+ */
+const crSearchPubmed = async ( term, opts ) => {
+  const doi = await crSearchForDoi(term);
+
+  const pmid = await getPmidFromDoi(doi);
+
+  if( pmid ){
+    return {
+      searchHits: [ pmid ],
+      count: 1,
+      query_key: '-1',
+      webenv: 'imitated_pubmed_format_from_crossref_query'
+    };
+  } else {
+    return eSearchPubmed(doi, opts);
+  }
+};
+
+/**
+ * Search for an article's DOI using Crossref
+ * @async
+ * @param {String} term The search term (e.g. article title)
+ * @returns {String} The DOI, null if not found
+ */
+const crSearchForDoi = async ( term ) => {
+  const params = {
+    'query.bibliographic': term
+  };
+
+  const crUrl = CROSSREF_WORKS_URL + '?' + queryString.stringify( params );
+
+  const crRes = await fetch(crUrl, {
+    method: 'GET'
+  });
+
+  const crJson = await crRes.json();
+
+  const doi = _.get(crJson, ['message', 'items', 0, 'DOI']);
+
+  return doi;
+};
+
+/**
+ * Use the PMC API to get the PMID for an article from the DOI.
+ * @async
+ * @param {String} doi The article's DOI
+ * @returns {String} The article's PMID, null if not found (i.e. doesn't exist or pubmed flakes out)
+ */
+const getPmidFromDoi = async ( doi ) => {
+  const params = {
+    format: 'json',
+    ids: doi,
+    tool: 'Biofactoid',
+    email: 'info@biofactoid.org'
+  };
+
+  const url = PMC_ID_API_URL + '?' + queryString.stringify( params );
+
+  const res = await fetch(url, {
+    method: 'GET'
+  });
+
+  const json = await res.json();
+
+  let pmid = _.get(json, ['records', 0, 'pmid']);
+
+  return pmid;
+};
```

Comment:
This service doesn't appear to work with anything from Nature or Science. We're better off using the original `eSearchPubmed` given a doi.

e.g. 10.1126/science.abc1932 , 10.1038/s41586-020-2335-4

Reply:
It seems that neither service gives good coverage on its own

Reply:
PMC queries are first, because they have reliable hits:  It queries based on the DOI field, whereas Eutils doen't always pick up the DOI properly.  I agree that PMC doesn't cast as wide a net as we'd like, but I think it's a more reliable first pass.

Reply:
> This service doesn't appear to work with anything from Nature or Science. We're better off using the original `eSearchPubmed` given a doi.
> 
> e.g. 10.1126/science.abc1932 , 10.1038/s41586-020-2335-4

Weird, now these do work with PMC?....

Reply:
I suspect they're scrambling to keep up with all the preprints, especially the COVID-19 ones.  Their DB seems to be changing frequently, sometimes improving things and sometimes adding some instability.

The Pubmed queries seem like they're a bit variable, but they'll also probably get better with time.  So an alternative to using Crossref, at least for now, could be to just increase the metadata cronjob frequency.

If we do go with the hybrid Crossref route, then it may be useful to have a final fallback case that searches Eutils by title:

- First, try PMC by DOI
- If PMC didn't work, then try Eutils by DOI
- If Eutils DOI didn't work, try Eutils by title

Reply:
I have some updates cooking here that more or less do what you said. Will push soon

Reply:
> > This service doesn't appear to work with anything from Nature or Science. We're better off using the original `eSearchPubmed` given a doi.
> > e.g. 10.1126/science.abc1932 , 10.1038/s41586-020-2335-4
> 
> Weird, now these do work with PMC?....

Today these don't work?